sympy/stats/drv_types.py

```diff
@@ -99,11 +102,10 @@ def DiscreteRV(symbol, density, set=S.Integers, **kwargs):
 
     """
     set = sympify(set)
-    pdf = Piecewise((density, set.as_relational(symbol)), (0, True))
-    pdf = Lambda(symbol, pdf)
     # have a default of False while `rv` should have a default of True
     kwargs['check'] = kwargs.pop('check', False)
-    return rv(symbol.name, DiscreteDistributionHandmade, pdf, set, **kwargs)
+    pmf = Lambda(symbol, density)
+    return rv(symbol, DiscreteDistributionHandmade, pmf, set, **kwargs)
```

Comment:
`symbol.name` also works but that removes the assumptions of the underlying `symbol` when args are re-sympified in `rv` function.